File: leetcode/3663.cpp
#include <print>

class Solution {
  public:
    int getLeastFrequentDigit(int n) {
        std::array<int, 10> cnts = {0};

        while (n > 0) {
            int digit = n % 10;
            n = n / 10;
            cnts[digit] += 1;
        }

        int lf_digit = 0;
        int minimum = 999;

        for (auto const [dig, cnt] : std::views::enumerate(cnts)) {
            if ((cnt != 0) && (cnt < minimum)) {
                lf_digit = dig;
                minimum = cnt;
            }
        }

        return lf_digit;
    }
};

File: leetcode/3663.kt
class Solution {
    fun getLeastFrequentDigit(n: Int): Int {
        val cnts: MutableMap<Int, Int> = mutableMapOf()
        var res: Int = n

        while (res > 0) {
            val dig: Int = res % 10
            res = res / 10
            cnts[dig] = cnts.getOrDefault(dig, 0) + 1
        }

        val minCnt: Int = cnts.values.minOrNull()!!

        for ((dig, cnt) in cnts.toList().sortedBy({ (dig, cnt) -> dig })) {
            if (cnt == minCnt) return dig
        }

        return -1
    }
}
